saged/sample_split_control.py

```diff
@@ -0,0 +1,217 @@
+                        help='The file to save the results to')
+    parser.add_argument('--neptune_config',
+                        help='A yaml formatted file containing init information for '
+                             'neptune logging')
+    parser.add_argument('--seed',
+                        help='The random seed to be used in splitting data',
+                        type=int,
+                        default=42)
+    parser.add_argument('--sample_split',
+                        help='If this flag is set, split cv folds at the sample level instead '
+                             'of the study level',
+                        action='store_true')
+
+    args = parser.parse_args()
+
+    with open(args.dataset_config) as data_file:
+        dataset_config = yaml.safe_load(data_file)
+
+    expression_df, sample_to_label, sample_to_study = utils.load_recount_data(args.dataset_config)
+    all_data = datasets.RefineBioMixedDataset(expression_df, sample_to_label, sample_to_study)
+
+    labeled_data = all_data.get_labeled()
+    labeled_data.subset_samples_to_labels(PREDICT_TISSUES)
+    labeled_data.recode()
+    label_encoder = labeled_data.get_label_encoder()
+
+    # Train the model on each fold
+    train_studies = []
+    train_sample_names = []
+    val_sample_names = []
+    train_sample_counts = []
+    subset_percents = []
+    val_losses = []
+
+    input_size = all_data[0].shape[0]
+    with open(args.model_config) as supervised_file:
+        model_config = yaml.safe_load(supervised_file)
+        model_config['input_size'] = input_size
+        # Output size is the same as the input because we're doing
+        # imputation
+        model_config['output_size'] = input_size
+
+    # Split the data into two sets
+    labeled_splits = labeled_data.get_cv_splits(num_splits=5,
+                                                seed=args.seed,
+                                                split_by_sample=args.sample_split)
+
+    del(all_data)
+
+    accuracies = []
+    balanced_accuracies = []
+    pretrain_sample_names = []
+    train_studies = []
+    train_sample_names = []
+    val_sample_names = []
+    train_sample_counts = []
+    subset_percents = []
+    pretrained_or_trained = []
+
+    for i in range(len(labeled_splits)):
+        neptune_run = None
+        # Parse config file
+        if args.neptune_config is not None:
+            with open(args.neptune_config) as neptune_file:
+                neptune_config = yaml.safe_load(neptune_file)
+                neptune_run = utils.initialize_neptune(neptune_config)
+
+        model_type = model_config['name']
+        SupervisedClass = getattr(models, model_type)
+        root_model = SupervisedClass(**model_config)
+
+        pretrain_base = copy.deepcopy(root_model)
+
+        # Get the pretraining fraction of the data
+        DatasetClass = type(labeled_data)
+        pretrain_list = [labeled_splits[(i+1) % 5], labeled_splits[(i+2) % 5]]
```

Comment:
Do you want to hard-code 5 here, or could you make this an argument to the script?

Reply:
I think I'll leave it hardcoded now then maybe update it during refactoring. It's hard to generalize a 2:2:1 ratio to arbitrary input, though to be fair it's only 2:2:1 to parallel the 20% val set other scripts, and they take arbitrary numbers of splits so :shrug: